packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -395,6 +433,11 @@ internal MethodProvider BuildJsonModelCreateCoreMethod()
             {
                 modifiers = MethodSignatureModifiers.Protected | MethodSignatureModifiers.Override;
             }
+
+            var typeOfT = GetModelArgumentType(_jsonModelTInterface);
+            // return type of this method may not be T, because this could be an override method
+            var returnType = GetOverriddenMethodReturnType(_baseSerializationProvider, PersistableModelCreateCoreMethodName) ?? typeOfT;
```

Comment:
we are doing this in a few places. Do you think it would be better if `GetOverriddenMethodReturnType` always returned the return type ?

Reply:
then its functionality does not match its name. The name says that this method should return the "the return type of the overridden method (base method)", therefore if there was one, it returns the return type of that method, returns null if there was not a base method.
Maybe the word "overridden" is misleading, I mean "the base method which gets overridden".
We could change the name to `GetJsonModelCoreReturnType` for what you described, but I kind of do not like it because the logic is not as intuitive as this.